examples/lm1b/train.py

```diff
@@ -527,10 +556,11 @@ def encode_strings(strs, max_len):
 
       # Shard data to devices and do a training step.
       with jax.profiler.StepTraceAnnotation("train", step_num=step):
-        batch = common_utils.shard(
-            jax.tree_util.tree_map(np.asarray, next(train_iter))
+        batch = next(train_iter)
+        batch = jax.tree_map(lambda x: jnp.array(x), batch)
```

Comment:
here you cast to jax array where the original seems to use numpy arrays, why did you do this?

Reply:
I get this error without casting:
```
File "/home/marcuschiam/flax/examples/lm1b/train.py", line 565, in train_and_evaluate
    state, metrics = jit_train_step(
TypeError: dtype tf.int32 not understood
```
The original element type is `EagerTensor` I think, with dtype `tf.int32`.

Reply:
Ok makes sense!